src/Shared/FileMatcher.cs

```diff
@@ -54,6 +54,29 @@ internal class FileMatcher
 
         private readonly GetFileSystemEntries _getFileSystemEntries;
 
+        private static class FileSpecRegexParts
+        {
+            internal const string BeginningOfLine = "^(?<FIXEDDIR>";
+            internal const string FixedDirWildcardDirSeparator = ")(?<WILDCARDDIR>";
```

Comment:
Instead of appending `)(`, why not have the previous phases add the closing parenthesis themselves?

Reply:
Yes, looks more clear. Since the stringbuilder is hopefully presized, shouldn't be that much of a hit. You could also call `sb.EnsureCapacity(sb.length + WildCardGroupStart.Length + GroupEnd.Start)` to control eventual resizes.